codex/blockexchange/engine/engine.nim

```diff
@@ -529,8 +529,13 @@ proc taskHandler*(b: BlockExcEngine, task: BlockExcPeerCtx) {.gcsafe, async.} =
 
   trace "wantsBlocks", peer = task.id, n = wantsBlocks.len
   if wantsBlocks.len > 0:
-    trace "Got peer want blocks list", items = wantsBlocks.len
-
+    # Consider only the wants not currently in-flight,
+    # Then mark them as in-flight.
+    wantsBlocks.keepItIf(not it.inFlight)
+    let wantAddresses = wantsBlocks.mapIt(it.address)
+    for peerWant in task.peerWants.mitems:
+      if peerWant.address in wantAddresses:
+        peerWant.inFlight = true
```

Comment:
I think this impl is bugged. 

1. All blocks that were not in flight are marked as in flight.
2. A block that was marked as `inFlight=true` was not found in repostore.
3. Such block will never be delivered (even if we'll have it in repostore future).

I liked previous impl, since it was very simple and achieved the same result.

Reply:
Good catch! I don't see how to detect if transmit failed. The send method doesn't return anything. I can set the flag back to false when the local read fails.

Reply:
It seems I can't figure out if the block transmit failed without changing the signatures all over the codebase. Also I can't write any unit tests for this because everything's so tightly coupled. I can check it's OK using the integration/dist-tests, but that it can't check that transmit-failures or local-read failures are handled correctly. At least, not without running the tests for a very long time and hoping one of those things happen.